jakarta-eclipse/src/org/jakarta/lsp4e/JakartaLanguageClient.java

```diff
@@ -78,4 +90,30 @@ public boolean isCanceled() {
  			return JDTServicesManager.getInstance().getExistingContextsFromClassPath(uri, snippetContexts);
  		});
  	}
+ 	
+	@Override
+	public CompletableFuture<List<Either<Command, CodeAction>>> getCodeAction(CodeActionParams params) {
+		return codeAction(params)
+				.thenApply(codeActions ->{
+					return codeActions.stream()
+							.map(ca -> {
+								Either<Command, CodeAction> e = Either.forRight(ca);
+								return e;
+							}.collect(Collectors.toList()));
+				});
+	}
```

Comment:
This method should look like this, and you can remove the helper `codeAction` method below:
```java
@Override
public CompletableFuture<List<CodeAction>> getCodeAction(CodeActionParams params) {

	JDTUtils utils = new JDTUtils();
	JakartaJavaCodeActionParams jakartaParams = new JakartaJavaCodeActionParams(params.getTextDocument(),
			params.getRange(), params.getContext());

	return CompletableFutures.computeAsync((cancelChecker) -> {
		IProgressMonitor monitor = getProgressMonitor(cancelChecker);
		try {
			return (List<CodeAction>) JDTServicesManager.getInstance().getCodeAction(jakartaParams, utils, monitor);
		} catch (JavaModelException e) {
			return Collections.emptyList();
		}
	});
}
```